### ISSUE TITLE
Error updating lookup field from primary key field

### ISSUE THREAD

**Issue by MarkMpn** [MarkMpn/Sql4Cds#11]
```sql
UPDATE a
SET    primarycontactid = c.contactid
FROM   account AS a
       INNER JOIN
       contact AS c
       ON a.accountid = c.parentcustomerid;
```

**Comment by malam-dev** [comment]
I ended up changing the `UpdateQuery` `ExecuteInternal` on my local copy with: 

```
                foreach (var attr in Updates)
                {
                    object value = attr.Value(entity);
                    if (value is Guid?)
                    {
                        var attrMetadata = meta.Attributes.SingleOrDefault(a => a.LogicalName == attr.Key);
                        if (attrMetadata != null && attrMetadata.AttributeType == AttributeTypeCode.Lookup)
                        {
                            value = new EntityReference(((LookupAttributeMetadata)attrMetadata).Targets[0], (Guid)value);
                        }
                    }
                    update[attr.Key] = value;
                }
```

**Comment by MarkMpn** [comment]
Released in 2.2.0
